pacta/email.go

```diff
@@ -0,0 +1,53 @@
+package pacta
+
+import (
+	"fmt"
+	"strings"
+	"unicode"
+	"unicode/utf8"
+
+	normalizer "github.com/dimuska139/go-email-normalizer"
+)
+
+var n = normalizer.NewNormalizer()
+
+func CanonizeEmail(email string) (string, error) {
+	// Performs basic validation, preventing obviously malformed email addresses, but
+	// not capturing the entirety of the RFC 5322 grammar.
+	split := strings.Split(email, "@")
+	if len(split) != 2 {
+		return "", fmt.Errorf("invalid email, wrong number of at-signs: %q", email)
+	}
+	if !isASCII(split[0]) || !utf8.ValidString(split[0]) {
+		return "", fmt.Errorf("invalid email, non-ASCII or UTF-8 local part: %q", split[0])
+	}
+	if includesSpace(email) {
+		return "", fmt.Errorf("invalid email, whitespace in email address: %q", split[0])
+	}
```

Comment:
I feel like this is one of those all downside, no upside checks and we should just do email verification (which I think we already do with Azure). Since [(quoted) spaces are allowed by the email spec](https://en.wikipedia.org/wiki/Email_address#Local-part), not allowing them is just guaranteeing we make _someone's_ life very annoying for no reason.

Reply:
SG